### PR TITLE
gio: Implement `Extend` on `ListStore` for `AsRef<Object>`

### PR DESCRIPTION
**Pull request by sdroege**
This also allows to use iterators over references of objects.

Also add a test for `extend()`.

----

See https://github.com/gtk-rs/gtk-rs-core/commit/277b94f0edecbee063f2ad12093e346868c8f8ef#r62142353

CC @ranfdev

### CODE REVIEW

File: gio/src/list_store.rs [hunk 1]
```diff
@@ -100,4 +104,17 @@ mod tests {
...
         assert_eq!(list.item(1), Some(item1.upcast()));
     }
+
+    #[test]
+    fn extend() {
+        let item0 = ListStore::new(ListStore::static_type());
+        let item1 = ListStore::new(ListStore::static_type());
+        let mut list = ListStore::new(ListStore::static_type());
+        list.extend(&[&item0, &item1]);
+        assert_eq!(list.item(0).as_ref(), Some(item0.upcast_ref()));
```
Comment (ranfdev) [inline gio/src/list_store.rs:114]: You could remove `.as_ref()` and `.upcast_ref()` and it would work equally well

Comment (ranfdev) [inline gio/src/list_store.rs:114]:
```suggestion
        assert_eq!(list.item(0), Some(item0.upcast()));
```

Comment (sdroege) [reply gio/src/list_store.rs:114]: No, because `upcast()` consumes the value and then it's gone and can't be used for the following comparison :)

Comment (ranfdev) [reply gio/src/list_store.rs:114]: Ups, sorry. 

I don't have an "approve" button, but i approve this PR.

**Review by GuillaumeGomez — APPROVED** [review]

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**